pipeline/recorder/src/main/scala/uk/ac/wellcome/platform/recorder/Implicits.scala

```diff
@@ -0,0 +1,17 @@
+package uk.ac.wellcome.platform.recorder
+
+import io.circe.generic.semiauto._
+import io.circe._
+
+import uk.ac.wellcome.models.work.internal.TransformedBaseWork
+import uk.ac.wellcome.storage.streaming.Codec
+import uk.ac.wellcome.json.JsonUtil._
+
+object Implicits {
+
+  // Cache these here to improve compilation times (otherwise they are
+  // re-derived every time they are required)
+  implicit val workEncoder: Encoder[TransformedBaseWork] = deriveEncoder
+  implicit val workDecoder: Decoder[TransformedBaseWork] = deriveDecoder
+  implicit val workCodec: Codec[TransformedBaseWork] = Codec.typeCodec
+}
```

Comment:
Nice!

Could these be put in the common lib, not just in the recorder, for even bigger compile-time wins?

Reply:
Thanks for the review. I have done another PR for improving the compile times of all pipeline apps: https://github.com/wellcometrust/catalogue/pull/135